### PR TITLE
create e2e test environment

### PR DESCRIPTION
**Pull request by mouseless-eth**
Goal is to have an as close to mainnet environment as possible. The e2e tests are focused on treating the bundler as a blackbox when testing its endpoints. 

Went with docker-compose instead of the bash script because was too hard to orchestrate everything through bash.

For the e2e environment, it will spawn a local anvil + alto instance (separate docker containers), the EntryPoint and SimpleAccountFactroy will be deployed to the same address as mainnet (because of create2).

The e2e test cases are written in jest. And will be part of the CI pipeline.

Right now there is only one test for the eth_sendUserOperation endpoint, will add more later tests later

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,28 +12,29 @@
...
         "lint": "rome check ./",
         "lint:fix": "rome check ./ --apply",
         "format": "biome format . --write"
     },
     "devDependencies": {
-        "@biomejs/biome": "^1.5.0",
-        "@swc/core": "^1.3.56",
-        "@types/mocha": "^10.0.1",
+        "@biomejs/biome": "^1.5.1",
+        "@swc/core": "^1.3.102",
```
Comment (plusminushalf) [inline package.json:22]: not for you but I don't know why do we have this dependency, we don't use it anywhere.

**Review by plusminushalf — APPROVED** [review]

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    - main
+
+jobs:
+  docker:
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
```
Comment (nikmel2803) [inline .github/workflows/e2e.yaml:18]:
```suggestion
      uses: actions/checkout@v4
```

File: test/e2e/anvil/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,6 @@
+FROM ghcr.io/foundry-rs/foundry
```
Comment (nikmel2803) [inline test/e2e/anvil/Dockerfile:1]: why do we need this Dockerfile? you can run this image directly with docker compose

File: test/e2e/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+version: "2.2"
```
Comment (nikmel2803) [inline test/e2e/docker-compose.yml:1]:
```suggestion
version: "3.8"
```

File: test/e2e/anvil/Dockerfile (hunk 3 above)
Comment (nikmel2803) [reply test/e2e/anvil/Dockerfile:1]: i'd also recommend to specify the exact version for this dependency (not just `:latest`)

File: test/e2e/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+services:
+  anvil:
+    container_name: anvil
```
Comment (nikmel2803) [inline test/e2e/docker-compose.yml:5]: is there any specific reason to specify it? i think we can omit it

**Pull request merged**